### PR TITLE
P#55975 improve sorting controls 17550 3

### PR DESCRIPTION
**Pull request by hungnc89**

### CODE REVIEW

File: js/onoffice-sorting-selection.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+var onOffice = onOffice || {};
+jQuery(document).ready(function($) {
+    let sortingSelection = $("#viewrecordssorting").find("[name=oopluginlistviews-sortingtypedirection]").val();
+    console.log(sortingSelection)
```
Comment (jayay) [inline js/onoffice-sorting-selection.js:4]: forgotten debug output

File: plugin/Gui/AdminPageAddressListSettings.php [hunk 2]
```diff
@@ -127,20 +128,30 @@ private function addFormModelRecordsFilter()
...
 		$pFormModelFilterRecords->addInputModel($pInputModelFilter);
 		$pFormModelFilterRecords->addInputModel($pInputModelRecordCount);
-		$pFormModelFilterRecords->addInputModel($pInputModelSortBy);
-		$pFormModelFilterRecords->addInputModel($pInputModelSortOrder);
 		$this->addFormModel($pFormModelFilterRecords);
+
+
 	}
 
+	private function addFormModelRecordsSorting(){
```
Comment (jayay) [inline plugin/Gui/AdminPageAddressListSettings.php:142]: please check the indents here

File: plugin/Gui/AdminPageAddressListSettings.php [hunk 3]
```diff
@@ -172,6 +183,9 @@ protected function generateMetaBoxes()
 
 		$pFormFilterRecords = $this->getFormModelByGroupSlug(self::FORM_VIEW_RECORDS_FILTER);
 		$this->createMetaBoxByForm($pFormFilterRecords, 'normal');
+
+        $pFormFilterRecords = $this->getFormModelByGroupSlug(self::FORM_VIEW_RECORDS_SORTING);
```
Comment (jayay) [inline plugin/Gui/AdminPageAddressListSettings.php:187]: Please check the indents

File: plugin/Installer/DatabaseChanges.php [hunk 4]
```diff
@@ -161,6 +161,7 @@ public function install()
 
 		if ($dbversion == 20) {
 			$this->updateCreateAddressFieldOfIntersetAndOwnerForm();
+            dbDelta( $this->getCreateQueryListviews() );
```
Comment (jayay) [inline plugin/Installer/DatabaseChanges.php:164]: Was there a change in the method `getCreateQueryListviews()` or what is the reason to have it run here again?

File: plugin/Installer/DatabaseChanges.php [hunk 5]
```diff
@@ -296,19 +297,27 @@ private function installDataQueryForms()
 				$template = $templatePathsForm;
 			}
 		}
-		$this->_pWPDB->insert(
-			$tableName,
```
Comment (jayay) [inline plugin/Installer/DatabaseChanges.php:300]: This attempt to make the default form better is nice, but the default form will have to be removed in another project, so I think this block should be reverted.

**Review by jayay — CHANGES_REQUESTED** [review]

File: plugin/Installer/DatabaseChanges.php (hunk 5 above)
Comment (hungnc89) [reply plugin/Installer/DatabaseChanges.php:300]: Ok, I rollback code, but it's will make failed run test

File: js/onoffice-sorting-selection.js (hunk 1 above)
Comment (hungnc89) [reply js/onoffice-sorting-selection.js]: i fixed

File: plugin/Gui/AdminPageAddressListSettings.php (hunk 2 above)
Comment (hungnc89) [reply plugin/Gui/AdminPageAddressListSettings.php:142]: i fixed

File: plugin/Installer/DatabaseChanges.php (hunk 4 above)
Comment (jayay) [reply plugin/Installer/DatabaseChanges.php:164]: This is still open

File: plugin/Installer/DatabaseChanges.php (hunk 5 above)
Comment (jayay) [reply plugin/Installer/DatabaseChanges.php:300]: If you check `TestClassDatabaseChanges`, methods `testInstall()` and `testInstallMigrationsDataSimilarEstates()`, there's a filter that catches `CREATE TABLE` queries. A similar one would be needed for `INSERT` queries.

**Review by jayay — COMMENTED** [review]
@hungnc89 Please check the indents again for all modified files

**Comment by hungnc89** [comment]
> @hungnc89 Please check the indents again for all modified files

Yes, I'm doing, I push many times but it's not changed, I don't know why, maybe I will change to new PR

**Comment by jayay** [comment]
@nglelinh 3fe5cf8 adds >40 modified files, that doesn't look right to me.

**Comment by hungnc89** [comment]
I change PR to #175

**Pull request closed without merging**